includes/class-wcsatt-scheme-prices.php

```diff
@@ -1,388 +1,484 @@
+
+		add_filter( 'woocommerce_available_variation', array( __CLASS__, 'filter_variation_data' ), 0, 3 );
+
+		do_action( 'wcsatt_add_price_filters' );
 	}
 
 	/**
-	 * Remove price filters after modifying child product prices depending on the per-product pricing option state, including any discounts defined at bundled item level.
+	 * Remove price filters.
 	 *
 	 * @return  void
 	 */
 	public static function remove_price_filters() {
 
-		if ( self::$price_overriding_scheme ) {
-
+		if ( WCS_ATT_Core_Compatibility::is_wc_version_gte_2_7() ) {
+			remove_filter( 'woocommerce_product_get_price', array( __CLASS__, 'filter_get_price' ), 0, 2 );
+			remove_filter( 'woocommerce_product_get_sale_price', array( __CLASS__, 'filter_get_sale_price' ), 0, 2 );
+			remove_filter( 'woocommerce_product_get_regular_price', array( __CLASS__, 'filter_get_regular_price' ), 0, 2 );
+			remove_filter( 'woocommerce_product_variation_get_price', array( __CLASS__, 'filter_get_price' ), 0, 2 );
+			remove_filter( 'woocommerce_product_variation_get_sale_price', array( __CLASS__, 'filter_get_sale_price' ), 0, 2 );
+			remove_filter( 'woocommerce_product_variation_get_regular_price', array( __CLASS__, 'filter_get_regular_price' ), 0, 2 );
+		} else {
 			remove_filter( 'woocommerce_get_price', array( __CLASS__, 'filter_get_price' ), 0, 2 );
 			remove_filter( 'woocommerce_get_sale_price', array( __CLASS__, 'filter_get_sale_price' ), 0, 2 );
 			remove_filter( 'woocommerce_get_regular_price', array( __CLASS__, 'filter_get_regular_price' ), 0, 2 );
-			remove_filter( 'woocommerce_variation_prices', array( __CLASS__, 'filter_get_variation_prices' ), 0, 3 );
+		}
 
-			do_action( 'wcsatt_remove_price_filters' );
+		remove_filter( 'woocommerce_get_variation_prices_hash', array( __CLASS__, 'filter_variation_prices_hash' ), 0, 2 );
+		remove_filter( 'woocommerce_variation_prices', array( __CLASS__, 'filter_get_variation_prices' ), 0, 2 );
 
-			self::$price_overriding_scheme  = false;
-			self::$price_overridden_product = false;
-		}
+		remove_filter( 'woocommerce_available_variation', array( __CLASS__, 'filter_variation_data' ), 0, 3 );
+
+		do_action( 'wcsatt_remove_price_filters' );
 	}
 
 	/**
-	 * Filter get_variation_prices() calls to take price overrides into account.
+	 * Add price html filters.
 	 *
-	 * @param  array                $prices_array
-	 * @param  WC_Product_Variable  $product
-	 * @param  boolean              $display
-	 * @return array
+	 * @return  void
 	 */
-	public static function filter_get_variation_prices( $prices_array, $product, $display ) {
+	public static function add_price_html_filters() {
 
-		$subscription_scheme = self::$price_overriding_scheme;
+		add_filter( 'woocommerce_get_price_html', array( __CLASS__, 'filter_get_price_html' ), 9999, 2 );
 
-		if ( $subscription_scheme && self::has_subscription_price_override( $subscription_scheme ) ) {
+		do_action( 'wcsatt_add_price_html_filters' );
```

Comment:
That's a good way to approach it. I was thinking we'd avoid having SATT installed if possible, but it makes more sense to have installed and exclusively responsible for backward compatibility rather than running that code in Subscriptions.

> I'd vote in favor of keeping all filter/meta names in SATT agnostic to the eventual "merge" with WCS, to avoid making any migration-related decisions now

Sounds good to me 👍